### ISSUE TITLE
Tale must be restarted to see new versions

### ISSUE THREAD

**Issue by craig-willis** [whole-tale/girderfs#14]
**Problem**
New versions do not appear in /versions mount until container is restarted.

**Steps to Reproduce**

1. Create tale
1. Launch tale
1. Edit something
1. Create version
1. Open terminal and ls ../versions
1. Restart tale
1. Open terminal and ls ../versions
1. Repeat steps 3-7

**Expected Results**
New versions are accessible from inside container without restarting

**Actual Results**
New versions are only accessible from inside container after restarting

**Comment by hategan** [comment]
I wonder if this is related to the modification time issue that you just filed with respect to image re-building. 

What I know for sure is that ls will fail updating if the modification time is not properly updated.

**Comment by craig-willis** [comment]
@hategan I don't think this is the problem. In this case, the image was successfully built (the workspace modification time is only used for subsequent builds). With this issue, =when a new version is added via the API, GirderFS is not propagating the new version to the container.

**Comment by hategan** [comment]
> GirderFS is not propagating the new version to the container.

GirderFS caches the /Versions directory. Every time it is accessed through FUSE, it checks, in sequence:
- if it's been recently loaded (i.e. loaded from Girder less than 1 second ago)
- if the modification time of /Versions as reported by Girder is the same as the modification time of the /Versions object fetched earlier

Only if both of the above tests fail, does GirderFS fetch a new listing of /Versions.
So, as I said, if the mtime is not being updated on the filesystem where /Versions is actually stored on the Girder side, `ls /Versions` will always return the same thing and never be updated.

**Comment by craig-willis** [comment]
OK, that makes sense.  However, this mtime is different than the mtime used for image builds.

**Comment by craig-willis** [comment]
Testing this a while back, I thought this had been fixed by https://github.com/whole-tale/virtual_resources/pull/10 but am again seeing the reported behavior.   When I create a new version while the container is running, it does not appear via `ls ../versions` until the container is restarted.

**Comment by craig-willis** [comment]
Noting that this is currently also the case with version rename and delete operations.

**Comment by Xarthisius** [comment]
The issue here is that listing of version root is associated in girderfs cache with the Tale object. I guess https://github.com/whole-tale/girderfs/blob/master/girderfs/core.py#L1256 is biting us here...

As workaround we could call `Tale().updateTale()` every time something happens in versionRoot folder.

**Comment by hategan** [comment]
@Xarthisius, maybe. I'm looking at the `rename()` code and I can't see anything that updates the root folder, so I'm not sure using the versionRoot folder as root in GirderFS would make a difference.

But it may be wise to add a sequence number (or equivalent) to the tale.

**Comment by Xarthisius** [comment]
> @Xarthisius, maybe. I'm looking at the `rename()` code and I can't see anything that updates the root folder, so I'm not sure using the versionRoot folder as root in GirderFS would make a difference.
> 
> But it may be wise to add a sequence number (or equivalent) to the tale.

Yeah, that was another thing we found today. I fixed that via https://github.com/whole-tale/wt_versioning/pull/23 Now everything i.e. version folder, versions root folder and a tale, should be updated during rename.
